test/functional/index.test.js

```diff
@@ -1354,4 +1356,101 @@ describe('Indexes', function() {
       });
     }
   });
+
+  context('should throw an error if commitQuorum specified on MongoDB < 4.4', function() {
+    function throwErrorTest(testCommand) {
+      return {
+        metadata: { requires: { mongodb: '<4.4' } },
+        test: function(done) {
+          const client = this.configuration.newClient();
+          client.connect(err => {
+            expect(err).to.not.exist;
+            const db = client.db('test');
+            const collection = db.collection('commitQuorum');
+            testCommand(db, collection, (err, result) => {
+              expect(err.message).to.equal(
+                '`commitQuorum` option for `createIndexes` not supported on servers < 4.4'
+              );
+              expect(result).to.not.exist;
+              client.close(done);
+            });
+          });
+        }
+      };
+    }
+    it(
+      'db.createIndex',
+      throwErrorTest((db, collection, cb) =>
+        db.createIndex(collection.collectionName, 'a', { commitQuorum: 'all' }, cb)
+      )
+    );
+    it(
+      'collection.createIndex',
+      throwErrorTest((db, collection, cb) =>
+        collection.createIndex('a', { commitQuorum: 'all' }, cb)
+      )
+    );
+    it(
+      'collection.createIndexes',
+      throwErrorTest((db, collection, cb) =>
+        collection.createIndexes(
+          [{ key: { a: 1 } }, { key: { b: 1 } }],
+          { commitQuorum: 'all' },
+          cb
+        )
+      )
+    );
+  });
+
+  context('should run command with commitQuorum if specified on MongoDB >= 4.4', function() {
+    function commitQuorumTest(testCommand) {
+      return {
+        metadata: { requires: { mongodb: '>=4.4', topology: ['replicaset', 'sharded'] } },
+        test: function(done) {
```

Comment:
Could this use `withMonitoredClient`?

Reply:
No point having those helpers if they don't get used when appropriate, I'll update that 👍